clients/fides-js/src/lib/i18n/index.ts

```diff
@@ -0,0 +1,92 @@
+import { setupI18n } from "./i18n-utils";
+
+/**
+ * Declare some generic i18n types that should be compatible with most i18n
+ * libraries. These are specifically modelled after LinguiJS, though!
+ *
+ * (see https://lingui.dev/ref/core#i18n)
+ */
+
+// Basic definition of a Locale string (e.g. "en" or "en-GB")
+type Locale = string;
+
+/**
+ * Basic definition of a i18n messages catalog. This works for a simple example like:
+ * ```
+ * {
+ *   "en": {
+ *     "experience.title": "Your Privacy Choices",
+ *     "experience.accept_button_label": "Accept",
+ *   },
+ *   "fr": {
+ *     "experience.title": "Vos choix en matière de confidentialité",
+ *     "experience.accept_button_label": "Accepter",
+ *   }
+ * }
+ * ```
+ */
+type Messages = Record<string, string>;
```

Comment:
Is it possible for this to be`Record<string, string | null | undefined>` ? When we `extractMessagesFromExperienceConfig`, for example, can we ever hit a state where a translation with a given key does not exist (like for optional fields on an experience translation).

Reply:
So, the *type* will never have undefined values, but it's definitely possible for a `i18n.t("some.key")` call to not match a string when it _thinks_ one should exist. So for example, something like `i18n.t("exp.accept_button_label")` isn't guaranteed to have a value.

Right now in that case it actually returns the key (e.g. `i18n.t("missing.key") === "missing.key"`) which'll look odd in the UI, so we'll need to check to see if we ever have cases where the components were relying on rendering empty strings... but I don't think so.

Reply:
Ah this makes sense 👍